x-pack/plugin/mapper-exponential-histogram/src/main/java/org/elasticsearch/xpack/exponentialhistogram/EncodedHistogramData.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.exponentialhistogram;
+
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.RamUsageEstimator;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.ByteArrayStreamInput;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.exponentialhistogram.ExponentialHistogram.MAX_SCALE;
+import static org.elasticsearch.exponentialhistogram.ExponentialHistogram.MIN_SCALE;
+
+/**
+ * Encodes the data of an exponential histogram in a compact format as byte array.
+ * Note that some data of exponential histograms is stored outside of this array as separate doc values for better compression,
+ * e.g. the zero threshold. This data is therefore not part of this encoding.
+ **/
+final class EncodedHistogramData {
+
+    /*
+     The encoding has the following format:
```

Comment:
What I would add here is some kind of encoding version, so that we can add new encoding schemes in the future and read multiple versions.

Reply:
But would you really mix those within the same segments/indices? Imo adding an encoding-version into the encoded `byte[]` would be very wasteful. If we add more encoding versions, it should either be part of the index/segment data (if this is supported) or an additional lucene field which then compresses nicely because it is usually the same value for all docs. And in those cases we can add it later when needed I would assume?